external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala

```diff
@@ -1254,6 +1254,9 @@ abstract class KafkaSourceSuiteBase extends KafkaSourceTest {
       .option("startingOffsets", s"earliest")
       .option("subscribe", topic)
       .load()
+      .selectExpr(
```

Comment:
It would be better if we check exhaustive columns here, so adding check for headers sounds better to me instead of limiting columns.

Reply:
I just indicated your comment that without `selectExpr` the test failed. Is it related to the addition of header field? I guess we should make sure it also works without `selectExpr` so that we can check also `headers` with memory sink.